diff --git a/dist/index.mjs b/dist/index.mjs
index 321bc4825371c718922f94537988e57c8f484b57..6bb72c3a2a3b428ac66944519f3126557bbac5bc 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -533,18 +533,19 @@ const resolveTypescriptProject = async (config, rootDir) => {
     ...parsedConfig?.options,
     ...config.compilerOptions
   };
+  const projectRootDir = parsedConfig?.options?.rootDir ?? rootDir;
   const projectFiles = parsedConfig ? [
     .../* @__PURE__ */ new Set([
       ...resolveProjectFilesFromParsedConfig(parsedConfig),
       ...referencedProjectMetadata.projectFiles
     ])
-  ].sort() : await resolveRootFilesFromGlobs(rootDir, includeArray, excludeArray);
+  ].sort() : await resolveRootFilesFromGlobs(projectRootDir, includeArray, excludeArray);
   const configFiles = resolveProjectConfigFiles(
     tsconfigPath,
     referencedProjectMetadata.configFiles
   );
   const rootFiles = parsedConfig ? await resolveDocgenRootFiles(
-    rootDir,
+    projectRootDir,
     includeArray,
     excludeArray,
     projectFiles
