diff --git a/internal/client/index.js b/internal/client/index.js
index bf05e5d3adf709985e28e76d9ed962359c2dccc9..19fdaa10d1425a742d05aae90dbc1c02fc0eb0f2 100644
--- a/internal/client/index.js
+++ b/internal/client/index.js
@@ -654,6 +654,11 @@ const addStyle = (styleContainerNode, cmpMeta, mode, hostElm) => {
                     if (BUILD.hydrateServerSide || BUILD.hotModuleReplacement) {
                         styleElm.setAttribute(HYDRATED_STYLE_ID, scopeId);
                     }
+                    const nonceElm = styleContainerNode.ownerDocument.querySelector('style[nonce]');
+                    if (nonceElm) {
+                      styleElm.setAttribute('nonce', nonceElm.getAttribute('nonce'));
+                      console.log(`CSP nonce-${nonceElm.getAttribute('nonce')}`);
+                    }
                     styleContainerNode.insertBefore(styleElm, styleContainerNode.querySelector('link'));
                 }
                 if (appliedStyles) {
@@ -2792,6 +2797,10 @@ const bootstrapLazy = (lazyBundles, options = {}) => {
     if (BUILD.invisiblePrehydration && (BUILD.hydratedClass || BUILD.hydratedAttribute)) {
         visibilityStyle.innerHTML = cmpTags + HYDRATED_CSS;
         visibilityStyle.setAttribute('data-styles', '');
+        const nonceElm = visibilityStyle.ownerDocument.querySelector('style[nonce]');
+        if (nonceElm) {
+          visibilityStyle.setAttribute('nonce', nonceElm.getAttribute('nonce'));
+        }
         head.insertBefore(visibilityStyle, metaCharset ? metaCharset.nextSibling : head.firstChild);
     }
     // Process deferred connectedCallbacks now all components have been registered
