diff --git a/internal/stencil-public-runtime.d.ts b/internal/stencil-public-runtime.d.ts
index 4e79baa330f81eadad96a2433a5357a5d9979524..fff8939dc8fd5c5319f92f78b179b2a266336ce3 100644
--- a/internal/stencil-public-runtime.d.ts
+++ b/internal/stencil-public-runtime.d.ts
@@ -610,7 +610,7 @@ export declare namespace JSXBase {
         code: JSXBase.HTMLAttributes;
         col: JSXBase.ColHTMLAttributes<HTMLTableColElement>;
         colgroup: JSXBase.ColgroupHTMLAttributes<HTMLTableColElement>;
-        data: JSXBase.HTMLAttributes<HTMLDataElement>;
+        data: JSXBase.DataHTMLAttributes<HTMLDataElement>;
         datalist: JSXBase.HTMLAttributes<HTMLDataListElement>;
         dd: JSXBase.HTMLAttributes;
         del: JSXBase.DelHTMLAttributes<HTMLModElement>;
@@ -824,6 +824,9 @@ export declare namespace JSXBase {
     interface ColgroupHTMLAttributes<T> extends HTMLAttributes<T> {
         span?: number;
     }
+    interface DataHTMLAttributes<T> extends HTMLAttributes<T> {
+        value?: string;
+    }
     interface DetailsHTMLAttributes<T> extends HTMLAttributes<T> {
         open?: boolean;
         onToggle?: (event: Event) => void;
@@ -1233,6 +1236,7 @@ export declare namespace JSXBase {
         tabIndex?: number;
         tabindex?: number | string;
         title?: string;
+        translate?: string;
         popover?: string | null;
         inputMode?: string;
         inputmode?: string;
